Update changelog with notes for v1.4.4 (#3480)
spiffe/spire: CHANGELOG.md

# Changelog

## [1.4.4] - 2022-10-05

### Added

> - Experimental support to configure the maximum number of SVIDs to keep in the agent's cache (#3181)

Reviewer: ```suggestion
- Experimental support for limiting the number of SVIDs in the agent's cache (#3181)
```

> - Support for k8s workload attestation when it is not possible to get the container ID (#3460)

Reviewer: I wonder if we should focus this sentence on the issue we fixed....
```suggestion
- Support for attesting Envoy proxy workloads when Istio is configured with holdApplicationUntilProxyStarts (#3460)
```

### Changed

> - Detect remote bundle endpoint misconfigurations (#3395)

Reviewer: ```suggestion
- Improved bundle endpoint misconfiguration diagnostics (#3395)
```
